src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1208 @@
+                xs = self.ln_post(xs)
+            else:
+                xs_ = []
+                for x, ln in zip(xs, self.ln_separate):
+                    x = ln(x)
+                    xs_.append(x)
+                xs = torch.stack(xs_, dim=0)  # shape = [layers, *, width, grid ** 2]
+            return xs
+        else:
+            x = self.transformer(x, x_mask)
+            x = x.permute(1, 0, 2)  # LND -> NLD
+            x = self.ln_post(x)
+            return x
+
+    def forward_pre(self, x: torch.Tensor):
+        x = self.conv1(x)  # shape = [*, width, grid, grid]
+        x = x.reshape(x.shape[0], x.shape[1], -1)  # shape = [*, width, grid ** 2]
+        x = x.permute(0, 2, 1)  # shape = [*, grid ** 2, width]
+        t = self.class_embedding.to(x.dtype) + torch.zeros(x.shape[0], 1, x.shape[-1], dtype=x.dtype, device=x.device)
+        x = torch.cat([t, x], dim=1)  # shape = [*, grid ** 2 + 1, width]
+        x = x + self.positional_embedding.to(x.dtype)
+        x = self.ln_pre(x)
+        x = x.permute(1, 0, 2)  # NLD -> LND
+        return x
+
+    def forward_post(self, x: torch.Tensor):
+        x = x.permute(1, 0, 2)
+        x = self.ln_post(x)
+        return x
+
+
+class BridgeTowerCLIP(nn.Module):
+    def __init__(
+        self,
+        embed_dim: int,
+        # vision
+        image_resolution: int,
+        vision_layers: Union[Tuple[int, int, int, int], int],
+        vision_width: int,
+        vision_patch_size: int,
+        transformer_width: int,
+        resolution_after=224,
+        model_type="bridgetower",
+        stop_gradient=False,
+        vit_layernorm_shared=True,
+        vit_remove_last=False,
+    ):
+        super().__init__()
+
+        vision_heads = vision_width // 64
+        self.visual = BridgeTowerVisualTransformer(
+            input_resolution=image_resolution,
+            patch_size=vision_patch_size,
+            width=vision_width,
+            layers=vision_layers,
+            heads=vision_heads,
+            output_dim=embed_dim,
+            resolution_after=resolution_after,
+            model_type=model_type,
+            stop_gradient=stop_gradient,
+            vit_layernorm_shared=vit_layernorm_shared,
+            vit_remove_last=vit_remove_last,
+        )
+
+        self.ln_final = BridgeTowerLayerNorm(transformer_width)
+
+        self.initialize_parameters()
+
+    def initialize_parameters(self):
+        proj_std = (self.visual.transformer.width**-0.5) * ((2 * self.visual.transformer.layers) ** -0.5)
+        attn_std = self.visual.transformer.width**-0.5
+        fc_std = (2 * self.visual.transformer.width) ** -0.5
+        for block in self.visual.transformer.resblocks:
+            nn.init.normal_(block.attn.in_proj_weight, std=attn_std)
+            nn.init.normal_(block.attn.out_proj.weight, std=proj_std)
+            nn.init.normal_(block.mlp.c_fc.weight, std=fc_std)
+            nn.init.normal_(block.mlp.c_proj.weight, std=proj_std)
```

Comment:
Can we do this from `_init_weights` function above? We prefer to have all weight initializations centralized in a single method. 
CLIP does it very nicely: https://github.com/huggingface/transformers/blob/1543cee7c8c95ef47f832b1f37625ba2923c4994/src/transformers/models/clip/modeling_clip.py#L411